### PR TITLE
*: goroutines use individual context

### PR DESCRIPTION
**Pull request by siddontang**
@coocood @disksing @nolouch

### CODE REVIEW

File: cmd/pd-tso-bench/main.go [hunk 1]
```diff
@@ -82,6 +82,9 @@ func main() {
 func showStats(ctx context.Context, durCh chan time.Duration) {
 	defer wg.Done()
 
+	statCtx, cancel := context.WithCancel(ctx)
+	defer cancel()
```
Comment (coocood) [inline cmd/pd-tso-bench/main.go]: Do we really need to call `cancel()`?
When `cancel` is called, no one is blocking on it.

Comment (siddontang) [inline cmd/pd-tso-bench/main.go]: The cancel function is for the statCtx resource, it is better to do it even it may has no effect.

**Comment by disksing** [comment]
LGTM.

**Comment by coocood** [comment]
LGTM

**Pull request merged**